app/serializers/api/v1/assignment_serializer.rb

```diff
@@ -9,5 +9,11 @@ class Api::V1::AssignmentSerializer
     (assignment.group.mentor_id == params[:current_user].id) || params[:current_user].admin?
   end
 
+  attributes :your_project_id do |assignment, params|
```

Comment:
This is a leaky abstraction, add the attribute `project_author_id` and find own project in client side or create a separate endpoint for this